address.go

```diff
@@ -119,6 +201,56 @@ func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
 	}
 }
 
+// decodeSegWitAddress parses a bech32 encoded segwit address string and
+// returns the witness version and witness program byte representation.
+func decodeSegWitAddress(hrp, address string) (byte, []byte, error) {
```

Comment:
my bad, after introducing the `IsBech32SegwitPrefix ` check, this should no longer be necessary.